styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerTest.java

```diff
@@ -328,34 +339,37 @@ public void shouldNotCleanupPodIfNotTerminated() {
     // inject mock status in real instance
     createdPod.setStatus(podStatus);
     when(podStatus.getContainerStatuses()).thenReturn(ImmutableList.of(containerStatus));
-    when(containerStatus.getName()).thenReturn(KubernetesDockerRunner.STYX_RUN);
+    when(containerStatus.getName()).thenReturn(CONTAINER_NAME);
     when(containerStatus.getState()).thenReturn(containerState);
 
     kdr.cleanupWithRunState(WORKFLOW_INSTANCE, name);
     verify(namedPod, never()).delete();
   }
 
   @Test
-  public void shouldNotCleanupPodIfNotStyxContainer() {
+  public void shouldNotCleanupNonStyxPod() {
     final String name = createdPod.getMetadata().getName();
     when(k8sClient.pods().withName(name)).thenReturn(namedPod);
     when(namedPod.get()).thenReturn(createdPod);
 
+    createdPod.getMetadata().setAnnotations(Collections.emptyMap());
+
     // inject mock status in real instance
     createdPod.setStatus(podStatus);
     when(podStatus.getContainerStatuses()).thenReturn(ImmutableList.of(containerStatus));
-    when(containerStatus.getName()).thenReturn("foo");
 
     kdr.cleanupWithRunState(WORKFLOW_INSTANCE, name);
     verify(namedPod, never()).delete();
   }
 
   @Test
-  public void shouldNotCleanupPodWithoutRunStateIfNotStyxContainer() {
+  public void shouldNotCleanupNonStyxPodWithoutRunState() {
     final String name = createdPod.getMetadata().getName();
     when(k8sClient.pods().withName(name)).thenReturn(namedPod);
     when(namedPod.get()).thenReturn(createdPod);
 
+    createdPod.getMetadata().setAnnotations(Collections.emptyMap());
+
     // inject mock status in real instance
     createdPod.setStatus(podStatus);
     when(podStatus.getContainerStatuses()).thenReturn(ImmutableList.of(containerStatus));
```

Comment:
Next line `when(containerStatus.getName()).thenReturn("foo");` could be deleted.